### DISCUSSION TITLE
[Feature] Enable comments & likes also for non system users

### DISCUSSION THREAD

**Discussion by 0xmex** [immich-app/immich#13034] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I, as a user, want to have the option to enable anonymous (or named with a name field) comments and like functions for external users which I share the album via link. Currently it is only possible via system users, as described in the docs: https://immich.app/docs/features/shared-albums/#shared-album-features

This would make communication with an external user (e.g. friend, model, client) much more easier.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by johnmmcgee** [comment]
I like this idea as well.

**Reply by bo0tzz** [reply]
@johnmmcgee if you don't have a substantial addition, just use the upvote button on the post please.

**Comment by jetmcquack** [comment]
I would like to add that this would be extremely useful for classrooms, closed groups with links , and many other use cases. 
Allowing comments also opens to spam, if the link is shared with wrong people, so take into account also this.

**Reply by johnmmcgee** [reply]
Easily solvable with the ability to allow approvals.

**Comment by tianxu0836** [comment]
This is an important feature for the photographer to work with their clients.

**Comment by rikk811** [comment]
Yes, I would love to have this feature! It would be great to know if it is being worked on. I understand the security concerns of a publicly shared album. But for sharing with frinends and family, it'd be great! It could even have an enable/disable option. 

I kind of think of it going this way:
The system could:

    Let anyone with the link click “Comment.”
    Prompt: “Enter your name to post comments.”
    Store the name with the comment, even without authentication
    Optionally use a cookie stored on the server so they don’t have to retype it.

My wife likes to share a lot of albums with friends. One thing she misses from using Google Photos is the ability for them to make comments and like/heart images. This would be a great feature!

**Reply by PseudoResonance** [reply]
I think with the current randomly generated share links, you've got plenty of entropy in that URL to keep it secure. It's effectively a password, but built into your URL and is seen on other online services, like Mega file sharing.

There's a little more security concerns with the URL being bookmarked, or automatically scraped from stolen text messages, stored in the browser history, etc but ultimately I don't think it's a huge concern. Not that it's a great way to think about security, but realistically I doubt many malicious people are even interested in this in the first place.

As for links intentionally being shared for malicious purposes, perhaps a moderation/approval system could be implemented too, or perhaps at a later date, since it's likely only useful to large, public instances.

**Comment by leonpawelzik** [comment]
Any update here?

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/
